File: src/python/spark-wiki-cs.py
clickstreamRaw = sqlContext.read \
  .format("com.databricks.spark.csv") \
  .options(header="true", delimiter="\t", mode="DROPMALFORMED", inferSchema="true") \
  .load("data/2015_02_clickstream.tsv")
  .cache()
  
# Converts the file to Parquet, an efficient data storage format. 
clickstreamRaw.write \
  .mode("overwrite") \
  .format("parquet") \
  .save("data/2015_02_clickstream.parquet")
  
clicks = sqlContext.read.parquet("data/2015_02_clickstream.parquet").cache()

# Calculate the number of clicks versus the number of Wikipedia clicks
all_clicks = clicks.selectExpr("sum(n) AS clicks").first().clicks
wiki_clicks = clicks.where("prev_id IS NOT NULL").selectExpr("sum(n) AS clicks").first().clicks
float(wiki_clicks) / all_clicks * 100

# Make clicks available as a SQL table.
clicks.registerTempTable("clicks")

File: src/scala/spark-wiki-cs.scala
import org.apache.spark._
import org.apache.spark.graphx._
import org.apache.spark.rdd.RDD
import org.apache.spark.graphx.lib.PageRank
import org.apache.spark.graphx.lib.ShortestPaths
import org.apache.spark.sql._
import org.apache.spark.sql.types.{StructType,StructField,StringType}

  val clickstreamDF = sqlContext.read.format("com.databricks.spark.csv")
    .option("header", "true")
    .option("delimiter", "\\t")
    .option("mode", "PERMISSIVE")
    .option("inferSchema", "true")
    .load("/Users/mbaker003c/spark/data/2015_02_clickstream.tsv")
    .where("n > 100")
    .cache()